src/EditorFeatures/Core/Implementation/LanguageServer/VisualStudioInProcLanguageServer.cs

```diff
@@ -35,71 +34,39 @@ internal class VisualStudioInProcLanguageServer : LanguageServerTarget
             ILspLogger logger,
             ImmutableArray<string> supportedLanguages,
             string? clientName,
-            WellKnownLspServerKinds serverKind)
-            : base(requestDispatcherFactory, jsonRpc, capabilitiesProvider, workspaceRegistrationService, lspMiscellaneousFilesWorkspace: null, globalOptions, listenerProvider, logger, supportedLanguages, clientName, serverKind)
-        {
-            _supportedLanguages = supportedLanguages;
-        }
-
-        public override Task InitializedAsync()
+            WellKnownLspServerKinds serverKind,
+            LspMiscellaneousFilesWorkspace? miscellaneousFilesWorkspace = null)
+            : base(requestDispatcherFactory, jsonRpc, capabilitiesProvider, workspaceRegistrationService, miscellaneousFilesWorkspace, globalOptions, listenerProvider, logger, supportedLanguages, clientName, serverKind)
         {
-            return Task.CompletedTask;
         }
 
-        [JsonRpcMethod(VSInternalMethods.DocumentPullDiagnosticName, UseSingleObjectParameterDeserialization = true)]
-        public Task<VSInternalDiagnosticReport[]?> GetDocumentPullDiagnosticsAsync(VSInternalDocumentDiagnosticsParams diagnosticsParams, CancellationToken cancellationToken)
-        {
-            Contract.ThrowIfNull(_clientCapabilities, $"{nameof(InitializeAsync)} has not been called.");
-
-            return RequestDispatcher.ExecuteRequestAsync<VSInternalDocumentDiagnosticsParams, VSInternalDiagnosticReport[]?>(
-                Queue, VSInternalMethods.DocumentPullDiagnosticName,
-                diagnosticsParams, _clientCapabilities, ClientName, cancellationToken);
-        }
+        internal TestAccessor GetTestAccessor() => new(this);
```

Comment:
the changes in this file are essentially removing the jsonrpcmethods (since we provide them at runtime) and adding a test accessor since we're now using this in tests.